test/runtests.jl

```diff
@@ -1059,3 +1059,8 @@ end
     end
     @test count(a) + count(!, a) == length(a)
 end
+
+@testset "unique" begin
+    a = AccessCountDiskArray((1:100) .& 7, chunksize=(9,))
+    @test length(unique(a)) == length(unique(identity, a)) == 8
```

Comment:
We need a test of the answer here too

Reply:
done.  i also added a test for iteration.

do you have any insight as to why is using `reduce` faster than `union`?  the mem alloc is the same.

Reply:
Not totally sure, I don't know exactly what the algs do internally. 

But `reduce` is only holding two active arrays at any one time, rather than all of them. And `union!` and `append!` are mostly working on the same block of memory so maybe better cache locality or something.

We could actually preallocate and reuse the memory for the reduce method to make it a little faster again, but probably not worth it at this stage